subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/cache/DefaultGeneralCompileCaches.java

```diff
@@ -97,4 +106,9 @@ public LocalJarClasspathSnapshotStore createLocalJarClasspathSnapshotStore(Strin
     public LocalClassSetAnalysisStore createLocalClassSetAnalysisStore(String taskPath) {
         return new LocalClassSetAnalysisStore(taskPath, taskCompileCache);
     }
+
+    @Override
+    public AnnotationProcessorPathStore createAnnotationProcessorPathStore(String taskpath) {
```

Comment:
We now have the compile information for a task spread across 3 stores, but we (I think) always load and store all 3 pieces of information, so there doesn't seem to be any benefit to this and only cost. I wonder if it would be better to merge these into a single store.

Reply:
Yep, we should.